Library/Homebrew/cmd/upgrade.rb

```diff
@@ -41,7 +42,10 @@ def upgrade
       puts pinned.map { |f| "#{f.full_name} #{f.pkg_version}" } * ", "
     end
 
-    outdated.each { |f| upgrade_formula(f) }
+    outdated.each do |f|
+      upgrade_formula(f)
```

Comment:
`f.installed?` will only return true when the **latest** version is installed which means upgrading is success.